ql/src/java/org/apache/hadoop/hive/ql/exec/tez/DagUtils.java

```diff
@@ -1221,9 +1221,9 @@ private Map<String, LocalResource> addTempResources(Configuration conf, String h
         LOG.info("Skipping vertex resource " + file + " that already exists in the session");
         continue;
       }
-      Path hdfsFilePath = new Path(hdfsDirPathStr, getResourceBaseName(new Path(file)));
-      LocalResource localResource = localizeResource(new Path(file),
-          hdfsFilePath, type, conf);
+      Path path = FileUtils.resolveSymlinks(new Path(file), conf);
```

Comment:
We do a ``new Path(file)`` above at L1220 as well, can we refactor it into a variable?
Secondly, we do have a ``null`` check for path inside FileUtils, not sure if it is of any use for us, since if ``file`` is ``null`` we won't pass ``null`` path but the code will throw an ``IAE``

Reply:
1. refactor to variable, yes 👍 
2. null check in the util: it's for general usage, to have to user get an IAE or NPE somewhere in the business logic instead of a utility method, I have no strong opinions about this though...I can refactor the util method to throw an IAE in case of null and tune the unit test accordingly